### DISCUSSION TITLE
Checkbox Counter in facets

### DISCUSSION THREAD

**Discussion by FrancescoBrunoDev** [vufind-org/vufind#3254] [Q&A] [answered]
Hello everyone,

I would like to know if there is a way similar to the one available in facets to have a counter. 

For example, I can do:

```
[Results]
online_available = translation
```

and this, like always, shows a counter. However,

```
[CheckboxFacets]
online_access: true = "ReadOnline"
```

does not display the corresponding counter.

Thank you for any feedback.

**Comment by demiankatz** [comment, marked as answer]
@FrancescoBrunoDev, the challenge here is that checkbox filters support complex queries, whereas other filters use simple values. Solr provides counts for basic facets, but in order to get counts for checkbox filters, we would need to perform additional Solr requests or restructure the way we are currently requesting the data from Solr. This is certainly not impossible, but it would come with performance costs.

I believe it is possible that someone somewhere has already implemented this as a local customization. I will highlight this discussion link on the vufind-tech mailing list to see if anyone there has anything to share.

I'm also open to adding this feature in the future if we can find a reasonably straightforward and efficient way to accomplish it.

**Reply by FrancescoBrunoDev** [reply]
Thank you @demiankatz for the prompt feedback. I understand the issue, and thanks for bringing it up on vufind-tech. Appreciate it!

**Comment by befehl** [comment]
As @demiankatz said, the count of Facets is included in the result of a search, whereras the checkbox filters are Filters which are set _before_ a search and therefore have no count. 
They also can be complex. Example: 
[CheckboxFacets]
format:Journal OR format:Newspaper OR format:Serial = "Periodicals only"
format:eBook OR format:Electronic OR format:Database = "Online resources only"

Afaik only solution is subsearches after each search.

**Reply by demiankatz** [reply]
I seem to recall that somebody might have implemented some solution that used AJAX calls to perform the checkbox queries asynchronously in order to fill in counts... but I may be misremembering, and as noted above, that obviously has performance implications. :-)

**Reply by befehl** [reply]
AJAX calls or anything else, a search is a search. 
One optimazation I can think of is to do a subsearch without asking for results or anythin else, just only the count of the search.

**Comment by EreMaijala** [comment]
> I seem to recall that somebody might have implemented some solution that used AJAX calls to perform the checkbox queries asynchronously in order to fill in counts... but I may be misremembering, and as noted above, that obviously has performance implications. :-)

That'd be me, I guess. :) References, with the caveat that this indeed only works for simple `field:value` type checkbox facets:

1. https://github.com/NatLibFi/NDL-VuFind2/blob/dev/module/Finna/src/Finna/AjaxHandler/GetSideFacets.php#L127
2. https://github.com/NatLibFi/NDL-VuFind2/blob/dev/module/Finna/src/Finna/AjaxHandler/GetSideFacets.php#L56
3. https://github.com/NatLibFi/NDL-VuFind2/blob/dev/module/Finna/src/Finna/Search/Solr/Params.php#L332

**Reply by FrancescoBrunoDev** [reply]
Hey @EreMaijala , thank you so much for sharing!
